packages/remix-dev/config/flat-routes.ts

```diff
@@ -316,3 +320,15 @@ function getRouteMap(
 
   return routeMap;
 }
+
+let routeRegex =
+  /(([+][/\\][^/\\:?*]+)|[/\\]((_index)|([^/\\:?*])))\.(tsx?|jsx?|mdx?)$$/;
+
+function isRouteModuleFile(filename: string) {
+  // flat files only need correct extension
+  let isFlatFile = !filename.includes(path.sep);
+  if (isFlatFile) {
+    return routeModuleExts.includes(path.extname(filename));
+  }
+  return routeRegex.test(filename);
+}
```

Comment:
nit: I think `filepath` might be a better name since a file name would never contain a path separator:

```suggestion
function isRouteModuleFile(filepath: string) {
  // flat files only need correct extension
  let isFlatFile = !filepath.includes(path.sep);
  if (isFlatFile) {
    return routeModuleExts.includes(path.extname(filepath));
  }
  return routeRegex.test(filepath);
}
```